simapp/app.go

```diff
@@ -140,8 +143,7 @@ func NewSimApp(
 ) *SimApp {
 
 	// TODO: Remove cdc in favor of appCodec once all modules are migrated.
-	cdc := std.MakeCodec(ModuleBasics)
-	appCodec := std.NewAppCodec(cdc)
+	appCodec, cdc := MakeCodecs()
```

Comment:
I don't see the need for an app codec anymore. We should just pass codec (which are defined in the respective modules) to the keeper constructors directly.

Reply:
Once we migrate the other modules this can just be a `Marshaler` instance instead. Note that I have added the `AnyUnpacker` interface to `Marshaler` now so that it is available through this interface.